daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1511,18 +1511,17 @@ private[lf] object Speedy {
     private[this] val actuals = machine.currentActuals
 
     // we must restore when catching a throw, or for normal execution
-    def restore(): Unit = {
+    def restore(machine: OnLedgerMachine): Unit = {
       machine.restoreBase(savedBase)
       machine.restoreFrameAndActuals(frame, actuals)
     }
 
-    override def execute(v: SValue): Control = {
-      restore()
-      machine.withOnLedger("KTryCatchHandler") { onLedger =>
-        onLedger.ptx = onLedger.ptx.endTry
+    override def execute(v: SValue): Control =
+      machine.asOnLedger("KTryCatchHandler") { machine =>
```

Comment:
Based on elsewhere, should `"KTryCatchHandler"` be replaced by `productPrefix`?

Reply:
ff155b94f8661cc0f28d255ae0cc0d8f525fb018